reports/report.md

```diff
@@ -4519,3 +4563,32 @@ A low level callee may consume all callers gas unexpectedly. Avoid unlimited imp
 
 
 
+## L-30: Function initializing state.
+
+Detects the immediate initialization of state variables through function calls that are not pure/constant, or that use non-constant state variable. Remove any initialization of state variables via non-constant state variables or function calls. If variables must be set upon contract deployment, locate initialization in the constructor instead.
+
+<details><summary>3 Found Instances</summary>
+
+
+- Found in src/FunctionInitializingState.sol [Line: 6](../tests/contract-playground/src/FunctionInitializingState.sol#L6)
+
+	```solidity
+	    uint public v = set(); // Initialize from function (sets to 77)
+	```
+
+- Found in src/FunctionInitializingState.sol [Line: 8](../tests/contract-playground/src/FunctionInitializingState.sol#L8)
+
+	```solidity
+	    uint public x = set(); // Initialize from function (sets to 88)
+	```
+
+- Found in src/FunctionInitializingState.sol [Line: 9](../tests/contract-playground/src/FunctionInitializingState.sol#L9)
+
+	```solidity
+	    uint public f = tes();
+	```
+
+</details>
```

Comment:
Here it is. It captures the variable declarations whose value is a function call *and*  the function definition makes references to non constant state variables. 

(This works regardless if it's marked as view or not.)